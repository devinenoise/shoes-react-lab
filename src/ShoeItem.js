import React, { Component } from 'react'
import { Link } from 'react-router-dom'


export default class ShoeItem extends Component {
    render() {

        const { data } = this.props;

        return (
            <div className="shoe-box">
                <li id="shoe-list">
                    <h1>{this.props.data.name}</h1>
                    <h3>brand: {this.props.data.brand}</h3>
                    <img className="shoe-images" src={this.props.data.url} alt={this.props.data.name} />
                    <h3>Type: {this.props.data.type}</h3>
                    <p>laces: {String(this.props.data.laces)}</p>
                    <Link to={`update/${data.id}`}> Update </Link>
                </li>
            </div>
        )
    }
}
