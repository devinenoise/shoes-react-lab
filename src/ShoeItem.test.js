import React from "react";
import ShoeItem from "./ShoeItem.js";
import renderer from "react-test-renderer";
import { Link } from "react-router-dom";


test('renders ShoeItem.js correctly', () => {
    const tree = renderer
        .create(<ShoeItem data={[]} /> <Link ></Link>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});