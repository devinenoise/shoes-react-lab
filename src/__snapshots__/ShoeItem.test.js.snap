// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App.js correctly 1`] = `
<div>
  <div
    className="header"
  >
    <h1>
      Shoe Shop
    </h1>
  </div>
  <a
    className="nav"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    className="nav"
    href="/AddShoe/New"
    onClick={[Function]}
  >
    Add A Shoe
  </a>
  <div>
    <ul />
  </div>
</div>
`;
